### ISSUE TITLE
Zookeeper installation does not honor install_dir attribute

### ISSUE THREAD

**Issue by dahui** [evertrue/exhibitor-cookbook#43]
My organization has a need to install Exhibitor and Zookeeper in a directory other than `/opt/`. While most of the facilities to do this are in both the Zookeeper and Exhibitor cookbooks, it appears that the Exhibitor cookbook is not correctly passing the custom installation path to the Zookeeper lwrp to install the software in `default.rb`. For now I have patched this locally and gotten it to work by including the `install_dir` directive in my local copy used to build our servers. Because of a legal NDA I signed within my organization, I am unable to submit pull requests to open source projects. 

The solve for this is to change the library call to the following:

```ruby
zookeeper node['exhibitor']['zookeeper_version'] do
  username  node['exhibitor']['user']
  user_home "/home/#{node['exhibitor']['user']}"
  install_dir node['zookeeper']['install_dir']
end
```
Thank you!

**Comment by claudobahn** [comment]
@jeffbyrnes I'm running into this, as well. The older cookbook used /opt/zookeeper/*, but the newer cookbook uses /opt/*, unfortunately this makes it tricky when sharing an S3 configuration.

So I'd like to just keep the /opt/zookeeper/* installation directory, but unfortunately can't, because https://github.com/evertrue/exhibitor-cookbook/blob/master/recipes/default.rb#L23 doesn't pass along the install_dir.

Any chance you could add `install_dir node['zookeeper']['install_dir']` as described above and push out a new version?

**Comment by jeffbyrnes** [comment]
Sorry folks, just now seeing this issue. I’m going to take a quick look right now, and see if I can get this out the door.

**Comment by jeffbyrnes** [comment]
Ok, I’ve refreshed myself on this now. So! The end result of this is that you should have an installation of ZooKeeper at `/opt/zookeeper-$version`, and a symlink to that dir at `/opt/zookeeper`. I updated the `zookeeper` cookbook this way to avoid `/opt/zookeeper/zookeeper-$version` being the install directory, because this is quite confusing to me.

So, when I test this cookbook, I get:

```shell
vagrant@default-ubuntu-1604:~$ ll /opt/
total 24
drwxr-xr-x  6 root      root      4096 Jul 26 14:10 ./
drwxr-xr-x 23 root      root      4096 Apr 27 18:05 ../
drwxrwxr-x  5 root      root      4096 Jul 26 14:10 chef/
drwx------  2 zookeeper root      4096 Jul 26 14:16 exhibitor/
drwxr-xr-x  9 root      root      4096 Apr 27 18:05 VBoxGuestAdditions-5.1.21/
lrwxrwxrwx  1 root      root        20 Jul 26 14:10 zookeeper -> /opt/zookeeper-3.4.9/
drwxr-xr-x 10 zookeeper zookeeper 4096 Jul 26 14:17 zookeeper-3.4.9/
```

@claudobahn, are you seeing that?

@dahui unfortunate that you cannot contribute; that‘s a rather counterproductive agreement your company has made you sign. Please pass along my displeasure at them that they will not allow you to help support tools they benefit from.

---

All that said, this is a legitimate feature request, and an easy one, and a non-breaking one.

I will caution, however, that `install_dir` is, perhaps, a misnomer (sorry about that). It ends up being used like so:

```ruby
  ark 'zookeeper' do
    url         "#{mirror}/zookeeper-#{new_resource.version}/zookeeper-#{new_resource.version}.tar.gz"
    version     new_resource.version
    prefix_root install_dir
    prefix_home install_dir
    checksum    new_resource.checksum if new_resource.checksum
    owner       username
    group       username
  end
```

The `ark` resource uses `install_dir` as the prefix for where it puts things, so you end up with `$install_dir/$name-$version` in the end. I couldn’t come up with a better name that wasn’t less confusing.

**Comment by jeffbyrnes** [comment]
All set, shipped `exhibitor` 4.1.0 to the Supermarket to add this feature. Thanks for bringing it to my attention!

**Comment by claudobahn** [comment]
@jeffbyrnes that is indeed what I get and it works fine for a fresh install.

Thanks for the fast turn-around!
